File: IMPERIUSDRACONIS/src/app/core/models/permisos.models.ts
import { CatalogItem } from './alumnos.models';

export interface PermisoDetalle {
  idPermiso: number;
  controlador: string;
  accion: string;
  tienePermiso: boolean;
}

export interface PermisoCargo {
  idCargo: number;
  cargoNombre: string;
  permisos: PermisoDetalle[];
}

export interface UpdatePermisoCargoRequest {
  permisos: Array<{
    idPermiso: number;
    tienePermiso: boolean;
  }>;
}

export interface CreatePermisoRequest {
  controlador: string;
  accion: string;
}

export type CargoOption = CatalogItem;

File: ImperiusDraconisAPI/ImperiusDraconisAPI/Models/Permisos/PermisosModels.cs
namespace ImperiusDraconisAPI.Models.Permisos;

public sealed class PermisoDetalleDto
{
    public int IdPermiso { get; init; }

    public string Controlador { get; init; } = string.Empty;

    public string Accion { get; init; } = string.Empty;

    public bool TienePermiso { get; init; }
}

public sealed class PermisoCargoDto
{
    public int IdCargo { get; init; }

    public string CargoNombre { get; init; } = string.Empty;

    public IReadOnlyCollection<PermisoDetalleDto> Permisos { get; init; } = Array.Empty<PermisoDetalleDto>();
}

public sealed class UpdatePermisoDetalleRequest
{
    public int IdPermiso { get; init; }

    public bool TienePermiso { get; init; }
}

public sealed class UpdatePermisoCargoRequest
{
    public IReadOnlyCollection<UpdatePermisoDetalleRequest> Permisos { get; init; } =
        Array.Empty<UpdatePermisoDetalleRequest>();
}

public sealed class CreatePermisoRequest
{
    public string Controlador { get; init; } = string.Empty;

    public string Accion { get; init; } = string.Empty;
}
